server/src/main/java/io/druid/client/ServerInventoryView.java

```diff
@@ -19,329 +19,12 @@
 
 package io.druid.client;
 
-import com.fasterxml.jackson.core.type.TypeReference;
-import com.fasterxml.jackson.databind.ObjectMapper;
-import com.google.common.base.Charsets;
-import com.google.common.base.Function;
-import com.google.common.base.Throwables;
-import com.google.common.collect.MapMaker;
-import com.metamx.emitter.EmittingLogger;
-import io.druid.concurrent.Execs;
-import io.druid.curator.inventory.CuratorInventoryManager;
-import io.druid.curator.inventory.CuratorInventoryManagerStrategy;
-import io.druid.curator.inventory.InventoryManagerConfig;
-import io.druid.java.util.common.StringUtils;
-import io.druid.java.util.common.lifecycle.LifecycleStart;
-import io.druid.java.util.common.lifecycle.LifecycleStop;
 import io.druid.timeline.DataSegment;
-import org.apache.curator.framework.CuratorFramework;
-
-import java.io.IOException;
-import java.nio.ByteBuffer;
-import java.nio.CharBuffer;
-import java.util.Map;
-import java.util.concurrent.ConcurrentMap;
-import java.util.concurrent.Executor;
-import java.util.concurrent.atomic.AtomicBoolean;
 
 /**
  */
-public abstract class ServerInventoryView<InventoryType> implements ServerView, InventoryView
+public interface ServerInventoryView extends ServerView, InventoryView
```

Comment:
moved the methods to InventoryView , however marker interface is still used to keep it configurable.